make/command

```diff
@@ -18,14 +22,17 @@ else
 BOOST_PROGRAM_OPTIONS_LIB=$(BOOST)/stage/lib/libboost_program_options.a
 BOOST_BUILD_OPTS=
 endif
+#ifeq ($(OS),Darwin)
+#CXXFLAGS_BOOST+=-fvisibility=hidden -fvisibility-inlines-hidden
+#endif
 
 .PRECIOUS: bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE)
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : CPPFLAGS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : LDFLAGS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : LDLIBS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : bin/%$(EXE) : bin/cmdstan/%.o $(BOOST_PROGRAM_OPTIONS_LIB)
 	@mkdir -p $(dir $@)
-	$(LINK.cpp) $^ $(BOOST_PROGRAM_OPTIONS_LIB) $(LDLIBS) $(OUTPUT_OPTION)
+	$(LINK.cpp) -fvisibility=hidden -fvisibility-inlines-hidden $^ $(BOOST_PROGRAM_OPTIONS_LIB) $(LDLIBS) $(OUTPUT_OPTION)
```

Comment:
Why did these go from only Darwin to always on? Are these just optims or is there a reason we need symbols to be hidden here?

Reply:
will change - that isn't correct and shouldn't have gotten mixed up with this PR.
fixed the makefile problem as part of PR #907